diff --git a/scripts/uosc.lua b/scripts/uosc.lua
index 2e2247d..68808b2 100644
--- a/scripts/uosc.lua
+++ b/scripts/uosc.lua
@@ -10,7 +10,7 @@ infinity = 1e309
 quarter_pi_sin = math.sin(math.pi / 4)
 
 -- Enables relative requires from `scripts` directory
-package.path = package.path .. ';' .. mp.find_config_file('scripts') .. '/?.lua'
+--package.path = package.path .. ';' .. mp.find_config_file('scripts') .. '/?.lua'
 
 require('uosc_shared/lib/std')
 
diff --git a/scripts/uosc_shared/elements/BufferingIndicator.lua b/scripts/uosc_shared/elements/BufferingIndicator.lua
index e2aa071..c1397e4 100644
--- a/scripts/uosc_shared/elements/BufferingIndicator.lua
+++ b/scripts/uosc_shared/elements/BufferingIndicator.lua
@@ -11,6 +11,7 @@ function BufferingIndicator:init()
 end
 
 function BufferingIndicator:decide_enabled()
+	if not state.is_stream then self.enabled = false return end
 	local cache = state.cache_underrun or state.cache_buffering and state.cache_buffering < 100
 	local player = state.core_idle and not state.eof_reached
 	if self.enabled then
diff --git a/scripts/uosc_shared/elements/PauseIndicator.lua b/scripts/uosc_shared/elements/PauseIndicator.lua
index 82a7e43..f900bc9 100644
--- a/scripts/uosc_shared/elements/PauseIndicator.lua
+++ b/scripts/uosc_shared/elements/PauseIndicator.lua
@@ -61,7 +61,7 @@ function PauseIndicator:render()
 	end
 
 	-- Icon
-	local size = round(math.min(display.width, display.height) * (is_static and 0.20 or 0.15))
+	local size = round(math.min(display.width, display.height) * 0.10)
 	size = size + size * (1 - self.opacity)
 
 	if self.paused then
diff --git a/scripts/uosc_shared/elements/Timeline.lua b/scripts/uosc_shared/elements/Timeline.lua
index 2ebcdf8..cb70c2f 100644
--- a/scripts/uosc_shared/elements/Timeline.lua
+++ b/scripts/uosc_shared/elements/Timeline.lua
@@ -83,6 +83,16 @@ function Timeline:get_time_at_x(x)
 	return state.duration * progress
 end
 
+local function show_clock(val)
+	if val then
+		if state.fullscreen then
+			mp.command("script-message-to clock show-clock yes")
+		end
+	else
+		mp.command("script-message-to clock show-clock no")
+	end
+end
+
 ---@param fast? boolean
 function Timeline:set_from_cursor(fast)
 	if state.time and state.duration then
@@ -120,8 +130,10 @@ function Timeline:on_prop_time() self:decide_enabled() end
 function Timeline:on_prop_border() self:update_dimensions() end
 function Timeline:on_prop_fullormaxed() self:update_dimensions() end
 function Timeline:on_display() self:update_dimensions() end
+function Timeline:on_mouse_enter() show_clock(true) end
 function Timeline:on_mouse_leave()
 	if not self.hovered_chapter then self:clear_thumbnail() end
+	show_clock(false)
 end
 function Timeline:on_global_mbtn_left_up()
 	if self.pressed then
@@ -208,7 +220,7 @@ function Timeline:render()
 	ass:new_event()
 	ass:pos(0, 0)
 	ass:append('{\\rDefault\\an7\\blur0\\bord0\\1c&H' .. bg .. '}')
-	ass:opacity(options.timeline_opacity)
+	ass:opacity((not state.pause or self:get_is_hovered()) and options.timeline_opacity or 0.45) -- might need function Timeline:on_prop_pause() request_render() end
 	ass:draw_start()
 	ass:rect_cw(bax, bay, fax, bby) --left of progress
 	ass:rect_cw(fbx, bay, bbx, bby) --right of progress
diff --git a/scripts/uosc_shared/elements/TopBar.lua b/scripts/uosc_shared/elements/TopBar.lua
index 977d5b8..8826231 100644
--- a/scripts/uosc_shared/elements/TopBar.lua
+++ b/scripts/uosc_shared/elements/TopBar.lua
@@ -83,7 +83,7 @@ function TopBar:update_dimensions()
 	self.size = state.fullormaxed and options.top_bar_size_fullscreen or options.top_bar_size
 	self.icon_size = round(self.size * 0.5)
 	self.spacing = math.ceil(self.size * 0.25)
-	self.font_size = math.floor((self.size - (self.spacing * 2)) * options.font_scale)
+	self.font_size = math.floor((self.size - (self.spacing * 1.6)) * options.font_scale)
 	self.button_width = round(self.size * 1.15)
 	self.ay = Elements.window_border.size
 	self.bx = display.width - Elements.window_border.size
@@ -132,17 +132,17 @@ function TopBar:render()
 		-- Playlist position
 		if state.has_playlist then
 			local text = state.playlist_pos .. '' .. state.playlist_count
-			local formatted_text = '{\\b1}' .. state.playlist_pos .. '{\\b0\\fs' .. self.font_size * 0.9 .. '}/'
+			local formatted_text = state.playlist_pos .. '{\\fs' .. self.font_size * 0.9 .. '}/'
 				.. state.playlist_count
-			local opts = {size = self.font_size, wrap = 2, color = fgt, opacity = visibility}
+			local opts = {size = self.font_size, wrap = 2, color = fg, opacity = visibility}
 			local bx = round(title_ax + text_width(text, opts) + padding * 2)
-			ass:rect(title_ax, title_ay, bx, self.by - bg_margin, {color = fg, opacity = visibility, radius = 2})
+			ass:rect(title_ax, title_ay, bx, self.by - bg_margin, {color = fgt, opacity = visibility, radius = 2})
 			ass:txt(title_ax + (bx - title_ax) / 2, self.ay + (self.size / 2), 5, formatted_text, opts)
-			title_ax = bx + bg_margin
+			title_ax = bx - 10
 		end
 
 		-- Title
-		local text = state.title
+		local text = state.title or 'mpv'
 		if max_bx - title_ax > self.font_size * 3 and text and text ~= '' then
 			local opts = {
 				size = self.font_size, wrap = 2, color = bgt, border = 1, border_color = bg, opacity = visibility,
diff --git a/scripts/uosc_shared/lib/menus.lua b/scripts/uosc_shared/lib/menus.lua
index e2a4ccc..4b62d5f 100644
--- a/scripts/uosc_shared/lib/menus.lua
+++ b/scripts/uosc_shared/lib/menus.lua
@@ -75,6 +75,7 @@ function create_select_tracklist_type_menu_opener(menu_title, track_type, track_
 		local first_item_index = #items + 1
 		local active_index = nil
 		local disabled_item = nil
+		local secondary_sid = nil
 
 		-- Add option to disable a subtitle track. This works for all tracks,
 		-- but why would anyone want to disable audio or video? Better to not
@@ -84,6 +85,7 @@ function create_select_tracklist_type_menu_opener(menu_title, track_type, track_
 		if track_type == 'sub' then
 			disabled_item = {title = 'Disabled', italic = true, muted = true, hint = '—', value = nil, active = true}
 			items[#items + 1] = disabled_item
+			secondary_sid = mp.get_property_native("secondary-sid", nil)
 		end
 
 		for _, track in ipairs(tracklist) do
@@ -105,7 +107,7 @@ function create_select_tracklist_type_menu_opener(menu_title, track_type, track_
 
 				items[#items + 1] = {
 					title = (track.title and track.title or 'Track ' .. track.id),
-					hint = table.concat(hint_values, ', '),
+					hint = (table.concat(hint_values, ', ')) .. ((track.selected and track.id == secondary_sid) and '²' or ''),
 					value = track.id,
 					active = track.selected,
 				}
diff --git a/scripts/uosc_shared/lib/utils.lua b/scripts/uosc_shared/lib/utils.lua
index d3f97fe..4868e18 100644
--- a/scripts/uosc_shared/lib/utils.lua
+++ b/scripts/uosc_shared/lib/utils.lua
@@ -129,6 +129,13 @@ function ass_escape(str)
 	return str
 end
 
+local function write_watch_later()
+    if mp.get_property_bool("save-position-on-quit") then
+        mp.command("script-message-to auto_save_state skip-delete-state")
+        mp.command("write-watch-later-config")
+    end
+end
+
 ---@param seconds number
 ---@return string
 function format_time(seconds)
@@ -355,6 +362,7 @@ function navigate_directory(delta)
 	local paths, current_index = get_adjacent_files(state.path, config.media_types)
 	if paths and current_index then
 		local _, path = decide_navigation_in_list(paths, current_index, delta)
+		if path then write_watch_later() end
 		if path then mp.commandv('loadfile', path) return true end
 	end
 	return false
@@ -365,6 +373,7 @@ function navigate_playlist(delta)
 	local playlist, pos = mp.get_property_native('playlist'), mp.get_property_native('playlist-pos-1')
 	if playlist and #playlist > 1 and pos then
 		local index = decide_navigation_in_list(playlist, pos, delta)
+		if index then write_watch_later() end
 		if index then mp.commandv('playlist-play-index', index - 1) return true end
 	end
 	return false
